### PR TITLE
React UI: Add better docs for the React UI

### PR DESCRIPTION
**Pull request by juliusv**
Signed-off-by: Julius Volz <julius.volz@gmail.com>

### CODE REVIEW

**Comment by juliusv** [comment]
@boyskila @hooten

File: web/ui/react-app/README.md [hunk 1]
```diff
@@ -1,68 +1,71 @@
...
+Due to a `"proxy": "http://localhost:9090"` setting in the `package.json` file, any API requests from the React UI are proxied to `localhost` on port `9090` by the development server. This allows you to run a normal Prometheus server to handle API requests, while iterating separately on the UI.
 
-## Learn More
+    [browser] ----> [localhost:3000 (dev server)] --(proxy API requests)--> [localhost:9090 (Prometheus)]
 
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
+## Running tests
 
-To learn React, check out the [React documentation](https://reactjs.org/).
+Create React App uses the [Jest](https://jestjs.io/) framework for running tests. To run tests in interactive watch mode:
```
Comment (boyskila) [inline web/ui/react-app/README.md]: Maybe adding coverage command is good too.

File: web/ui/react-app/README.md [hunk 2]
```diff
@@ -1,68 +1,71 @@
...
+* The [Node.js](https://nodejs.org/) JavaScript runtime.
+* The [Yarn](https://yarnpkg.com/) package manager.
+* *Recommended:* An editor with TypeScript, React, and [ESLint](https://eslint.org/) linting support. See e.g. [Create React App's editor setup instructions](https://create-react-app.dev/docs/setting-up-your-editor/). If you are not sure which editor to use, we recommend using [Visual Studio Code](https://code.visualstudio.com/docs/languages/typescript). Make sure that [the editor uses the project's TypeScript version rather than its own](https://code.visualstudio.com/docs/typescript/typescript-compiling#_using-the-workspace-version-of-typescript).
 
-### `npm run build`
+## Installing npm dependencies
 
-Builds the app for production to the `build` folder.<br>
-It correctly bundles React in production mode and optimizes the build for the best performance.
+The React UI depends on a large number of [npm](https://www.npmjs.com/) packages. These are not checked in, so you will need to download and install them locally via the Yarn package manager:
```
[the remark this answers is not in the document]
Comment (juliusv) [reply web/ui/react-app/README.md]: Yes, currently we should stick with Yarn due to the lockfile, but potentially we can switch back to npm at some point (read some blog posts mentioning that npm has pretty much caught up with Yarn in many ways, but is simpler).

File: web/ui/react-app/README.md [hunk 3]
```diff
@@ -1,68 +1,71 @@
...
+    yarn start
 
-Instead, it will copy all the configuration files and the transitive dependencies (Webpack, Babel, ESLint, etc) right into your project so you have full control over them. All of the commands except `eject` will still work, but they will point to the copied scripts so you can tweak them. At this point you’re on your own.
+This will open a browser window with the React app running on http://localhost:3000/. The page will reload if you make edits to the source code. You will also see any lint errors in the console.
 
-You don’t have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.
+Due to a `"proxy": "http://localhost:9090"` setting in the `package.json` file, any API requests from the React UI are proxied to `localhost` on port `9090` by the development server. This allows you to run a normal Prometheus server to handle API requests, while iterating separately on the UI.
```
[the remark this answers is not in the document]
Comment (juliusv) [reply web/ui/react-app/README.md]: I think that's best kept in the main README.md I think, where everyone will look first.

**Comment by boyskila** [comment]
In overall looks pretty good :)

File: web/ui/react-app/README.md (hunk 1 above)
Comment (juliusv) [reply web/ui/react-app/README.md]: Added coverage docs.

File: web/ui/react-app/src/ExpressionInput.test.tsx [hunk 4]
```diff
@@ -178,7 +178,7 @@ describe('ExpressionInput', () => {
       const downshift = expressionInput.find(Downshift);
       const instance: any = expressionInput.instance();
       const spyBlur = jest.fn();
-      instance.exprInputRef.current = { blur: spyBlur };
```
Comment (boyskila) [inline web/ui/react-app/src/ExpressionInput.test.tsx]: This change is part of https://github.com/prometheus/prometheus/pull/6316

Comment (boyskila) [reply web/ui/react-app/src/ExpressionInput.test.tsx]: Or its here as well because of failing build

Comment (juliusv) [reply web/ui/react-app/src/ExpressionInput.test.tsx]: Whoops shit, added it locally to make tests pass, will remove it from this PR.

**Pull request merged**